components/arm/client.go

```diff
@@ -8,31 +8,36 @@ import (
 	"github.com/edaniels/golog"
 	commonpb "go.viam.com/api/common/v1"
 	pb "go.viam.com/api/component/arm/v1"
+	robotpb "go.viam.com/api/robot/v1"
 	"go.viam.com/utils/protoutils"
 	"go.viam.com/utils/rpc"
 
 	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/spatialmath"
 )
 
-var errArmClientInputsNotSupport = errors.New("arm client does not support inputs directly")
+var errArmClientModelNotValid = errors.New("unable to retrieve a valid arm model from arm client")
 
 // client implements ArmServiceClient.
 type client struct {
 	name   string
 	conn   rpc.ClientConn
 	client pb.ArmServiceClient
+	model  referenceframe.Model
 	logger golog.Logger
 }
 
 // NewClientFromConn constructs a new Client from connection passed in.
 func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name string, logger golog.Logger) Arm {
 	c := pb.NewArmServiceClient(conn)
+	r := robotpb.NewRobotServiceClient(conn) // TODO(rb): remove the robot service client when we can use the framesystem as a dependency
 	return &client{
 		name:   name,
 		conn:   conn,
 		client: c,
+		model:  getModel(ctx, r, name),
```

Comment:
will the model ever need to be updated?

Reply:
A wrapper arm could have the model it points at change, and this client could point to a wrapper arm; so potentially.

But, I do not think that a wrapper arm actually supports having its model be updated without a restart.

Reply:
Yes the wrapper arm is only place this could conceivably happen but as Peter said, cannot be updated on the fly anyway.  Maybe this is something that we would want in the future so I could potentially change this, but I don't think its super necessary now.  Thoughts?

Reply:
It is certain we will want to eventually support this; if someone is developing an arm, we want them to be able to rapidly iterate on testing model files.

Let's make a ticket for supporting live updating of wrapper arms, and add a TODO comment here linking that ticket.

Reply:
maybe add a comment highlighting that this would be an issue if we ever do update the underlying frame without a restart

Reply:
Created DATA-853 and made a comment calling this out